.github/workflows/test.yml

```diff
@@ -1,18 +1,55 @@
+name: Test and build outline-cli
+
 on: [push, pull_request]
-name: Test
+
 jobs:
   test:
+    name: Test
     strategy:
       matrix:
         go-version: [1.19.x]
         os: [ubuntu-latest]
     runs-on: ${{ matrix.os }}
+
     steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
       - name: Install Go
         uses: actions/setup-go@v4
         with:
           go-version: ${{ matrix.go-version }}
+
+      - name: Get dependencies
+        run: go mod download
+
+      - name: Run tests with coverage
+        run: go test -coverprofile=coverage.txt -covermode=atomic ./...
+      
+      - name: Generate coverage report
+        run: go get github.com/axw/gocov/gocov
+        run: gocov convert coverage.txt
+
+  build:
+    name: Build (Main branch only)
+    strategy:
+      matrix:
+        go-version: [1.19.x]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    if: github.ref == 'refs/heads/main'  # This ensures the job runs only on the 'main' branch
```

Comment:
What about to introduce this as a second workflow where we have such a filter in the 'on' section instead? 🤔

Reply:
Makes sense, you mean a separate file, right?